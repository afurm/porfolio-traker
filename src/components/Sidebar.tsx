import React from 'react';
import Link from 'next/link';
import { Box, Typography, List, ListItem, ListItemIcon, ListItemText } from '@mui/material';
import { FiHome, FiBarChart2, FiDollarSign, FiSettings } from 'react-icons/fi';

const Sidebar: React.FC = () => {
  return (
    <Box
      sx={{
        width: 250,
        p: 2,
        bgcolor: 'background.paper',
        height: '100vh',
        boxShadow: 3,
      }}
    >
      <Box sx={{ mb: 4 }}>
        <Typography variant="h5" fontWeight="bold">
          CoinStats
        </Typography>
      </Box>
      <List>
        <Link href="/dashboard" passHref>
          <ListItem button component="a">
            <ListItemIcon>
              <FiHome />
            </ListItemIcon>
            <ListItemText primary="Dashboard" />
          </ListItem>
        </Link>
        <Link href="/analytics" passHref>
          <ListItem button component="a">
            <ListItemIcon>
              <FiBarChart2 />
            </ListItemIcon>
            <ListItemText primary="Analytics" />
          </ListItem>
        </Link>
        <Link href="/assets" passHref>
          <ListItem button component="a">
            <ListItemIcon>
              <FiDollarSign />
            </ListItemIcon>
            <ListItemText primary="Assets" />
          </ListItem>
        </Link>
        <Link href="/settings" passHref>
          <ListItem button component="a">
            <ListItemIcon>
              <FiSettings />
            </ListItemIcon>
            <ListItemText primary="Settings" />
          </ListItem>
        </Link>
      </List>
    </Box>
  );
};

export default Sidebar;
