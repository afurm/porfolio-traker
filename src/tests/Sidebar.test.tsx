import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Sidebar from '@/components/Sidebar';

describe('Sidebar Component Snapshot Test', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Sidebar />);
    expect(asFragment()).toMatchSnapshot();
  });
});
