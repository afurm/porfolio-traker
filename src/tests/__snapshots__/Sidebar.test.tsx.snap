// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Component renders correctly when user is authenticated 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1kfzhxo"
  >
    <div
      class="MuiBox-root css-1yjvs5a"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1yej70y-MuiTypography-root"
      >
        CoinStats
      </h5>
    </div>
    <ul
      class="MuiList-root MuiList-padding css-1u169s7-MuiList-root"
    >
      <a
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1qa1mqp-MuiListItem-root"
        href="/transaction-history"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="12"
              x2="12"
              y1="1"
              y2="23"
            />
            <path
              d="M17 5H9.5a3.5 3.5 0 0 0 0 7h5a3.5 3.5 0 0 1 0 7H6"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root css-1y2w5y4-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1edfpdg-MuiTypography-root"
          >
            Transaction History
          </span>
        </div>
      </a>
      <a
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1qa1mqp-MuiListItem-root"
        href="/add-transaction"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="12"
              x2="12"
              y1="1"
              y2="23"
            />
            <path
              d="M17 5H9.5a3.5 3.5 0 0 0 0 7h5a3.5 3.5 0 0 1 0 7H6"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root css-1y2w5y4-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1edfpdg-MuiTypography-root"
          >
            Add Transaction
          </span>
        </div>
      </a>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-193q0wg-MuiListItem-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-18ulbdb-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Logout
        </button>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`Sidebar Component renders correctly when user is not authenticated 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1kfzhxo"
  >
    <div
      class="MuiBox-root css-1yjvs5a"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1yej70y-MuiTypography-root"
      >
        CoinStats
      </h5>
    </div>
    <ul
      class="MuiList-root MuiList-padding css-1u169s7-MuiList-root"
    >
      <a
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1qa1mqp-MuiListItem-root"
        href="/transaction-history"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="12"
              x2="12"
              y1="1"
              y2="23"
            />
            <path
              d="M17 5H9.5a3.5 3.5 0 0 0 0 7h5a3.5 3.5 0 0 1 0 7H6"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root css-1y2w5y4-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1edfpdg-MuiTypography-root"
          >
            Transaction History
          </span>
        </div>
      </a>
      <a
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1qa1mqp-MuiListItem-root"
        href="/add-transaction"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="12"
              x2="12"
              y1="1"
              y2="23"
            />
            <path
              d="M17 5H9.5a3.5 3.5 0 0 0 0 7h5a3.5 3.5 0 0 1 0 7H6"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root css-1y2w5y4-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1edfpdg-MuiTypography-root"
          >
            Add Transaction
          </span>
        </div>
      </a>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-193q0wg-MuiListItem-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-18ulbdb-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Login
        </button>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
